peer/peer.go

```diff
@@ -1688,6 +1688,14 @@ cleanup:
 //
 // This function is safe for concurrent access.
 func (p *Peer) QueueMessage(msg wire.Message, doneChan chan<- struct{}) {
+	// Panic early on nil messages.  This provides a more useful stack
+	// trace to callers than hitting the panic in a long-lived peer
+	// goroutine that contains no information about what caller queued the
+	// nil message.
+	if msg == nil {
+		panic("peer: nil message")
```

Comment:
We discussed this before in a private chat, but just to mention it here: ignoring the message would be nice to do, but the semantics of whether the doneChan should be sent to or not is unclear if we simply ignore the message.  Continuing on in this state is probably going to create more issues than just panicing and fixing the bugs.

Reply:
To elaborate on what I am worried over, if we just ignore the message, I can see us getting into deadlock situations where no debug info is available at all (sans pprof).  This could occur both with, and without, the doneChan send, depending on how the caller is abusing the API.

Reply:
As a compromise, can we panic if doneChan != nil, and ignore and warn log with the stacktrace if it is nil?

Reply:
That sounds like a good compromise.  There are almost no cases of `done != nil` and there is definitely no ambiguity in the case of `done == nil` to worry about.